### PR TITLE
refactor: Rename environment variable file, remove support for multiple files

### PR DESCRIPTION
**Pull request by rhysbarrett**
### JIRA link

N/A

### Change description

This PR renames `.env.vagrant` as `.env.test` and removes support for loading of multiple environment files.

### Work checklist

- [x] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -27,7 +27,7 @@ WORKDIR /app
 
 COPY --from=prod-deps-image /build/node_modules/ ./node_modules
 COPY --from=build-image /build/dist/ ./dist
-COPY .env.vagrant ./
+COPY .env.test ./
```
Comment (geek-soft) [inline Dockerfile:30]: To be fair it should not be copied if this is just test file.

File: src/utils/ConfigLoader.ts [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+
+const DEFAULT_ENV_FILE = `.env`;
 
 export const loadEnvironmentVariablesFromFiles = () => {
-    dotenv.config({ path: DEFAULT_ENV_FILE });
-    if (process.env.NODE_ENV) {
-        const envFilePath = `${__dirname}/../../.env.${process.env.NODE_ENV}`;
-        dotenv.config({ path: envFilePath });
-    }
+    const env = getEnv('NODE_ENV');
```
Comment (geek-soft) [inline src/utils/ConfigLoader.ts:8]: I believe we should simplify it so only `.env.[env]` is loaded.

File: Dockerfile (hunk 1 above)
Comment (rhysbarrett) [reply Dockerfile:30]: Oh okay, cool. I'll remove.

File: src/utils/ConfigLoader.ts (hunk 2 above)
Comment (rhysbarrett) [reply src/utils/ConfigLoader.ts:8]: Okay, and we allow it to fail loudly if the environment variable isn't present?

File: src/utils/ConfigLoader.ts [hunk 3]
```diff
@@ -1,11 +1,9 @@
...
-const DEFAULT_ENV_FILE = `${__dirname}/../../.env`;
+import { getEnvOrThrow } from 'app/utils/EnvironmentUtils';
 
 export const loadEnvironmentVariablesFromFiles = () => {
-    dotenv.config({ path: DEFAULT_ENV_FILE });
-    if (process.env.NODE_ENV) {
-        const envFilePath = `${__dirname}/../../.env.${process.env.NODE_ENV}`;
-        dotenv.config({ path: envFilePath });
-    }
+    const env = getEnvOrThrow('NODE_ENV');
```
Comment (geek-soft) [inline src/utils/ConfigLoader.ts:6]: A `NODE_ENV` env is optional, using `getEnvOrThrow` will break application in cloud environments. Use more graceful option.

Comment (rhysbarrett) [reply src/utils/ConfigLoader.ts:6]: This seems to conflict with your previous comment: https://github.com/companieshouse/lfp-appeals-frontend/pull/259#discussion_r463078369

Either we assume NODE_ENV is present and break the application when it isn't, or we need a default/fallback file.

Comment (geek-soft) [reply src/utils/ConfigLoader.ts:6]: In previous comment I meant to stop loading `.env` and load `.env.[env]` but only when `NODE_ENV` is present. We always had conditional use of `NODE_ENV` and we should not change it.

**Pull request merged**